pilot/cmd/pilot-agent/main.go

```diff
@@ -151,9 +151,12 @@ var (
 	skipParseTokenEnv = env.RegisterBoolVar("SKIP_PARSE_TOKEN", false,
 		"Skip Parse token to inspect information like expiration time in proxy. This may be possible "+
 			"for example in vm we don't use token to rotate cert.").Get()
-	proxyXDSViaAgent = env.RegisterStringVar("PROXY_XDS_VIA_AGENT", "",
+	proxyXDSViaAgent = env.RegisterStringVar("ISTIO_META_PROXY_XDS_VIA_AGENT", "",
```

Comment:
long story. we tried to do bool. but enabling this in tests (see main_test.go) proved very difficult due to yaml quoting issues. Hence the string hack. This will go away once we have a proper mesh config variable for this

Reply:
if you want to avoid many checks you can just parse it to a bool right here at the global level